app/services/base_auth_service.rb

```diff
@@ -25,10 +54,18 @@ def validate_expected_fields!(decoded_token)
       end
     end
 
-    def setup_user(remote_user_id, user_atts)
-      user = User.where(primary_key => remote_user_id).first_or_initialize
-      user.update!(user_atts)
-      user
+    def setup_user(remote_user_id, user_atts, create_user)
+      users = User.where(primary_key => remote_user_id)
+
+      user_existence_validation(create_user, users)
+      user = existent_user(create_user, users)
+      return user if user
+
+      if create_user.nil?
```

Comment:
create_user is a boolean, right? So technically, create_user could be false, and false.nil? will evaluate to false. Why not put the if structure with the positive case first, so `if create_user`... 

I'm also kind of confused by this just in general. So if you *don't* want to create the user, first_or_initialize_user will still create a new record.... ?

I think having these helper methods (first_or_create_user and new_user) makes reading it more complex; my gut feeling is these could be done inline here and it would avoid having to mentally jump to another method to follow the code.

Reply:
Oh, I see! A comment to that effect would probably be sufficient then.

Reply:
@andrewroth I reworked the logic